File: string_processing/pl/string_joins_02.pl
#!/usr/bin/perl

########################################
# PL joins on ranges
# author: vladimir kulyukin
########################################

use strict;
use warnings;
use 5.10.0;

my @range1 = (1 .. 5);

sub join_number_range {
    my ($separator, $range) = @_;
    return join($separator, @{$range});
}

sub range_tests {
    print join_number_range('*',    \@_), "\n";
    print join_number_range(' ** ', \@_), "\n";
    print join_number_range('//',   \@_), "\n";
}

## a different version of join_number_range2 that 
## uses shift();
sub join_number_range2 {
    my $separator = shift();
    return join($separator, @_);
}

sub range_tests2 {
    print join_number_range2('*',    @_), "\n";
    print join_number_range2(' ** ', @_), "\n";
    print join_number_range2('//',   @_), "\n";
}

range_tests(@range1);
range_tests2(@range1);

File: string_processing/py/string_joins_02.py
#!/usr/bin/python

####################################
# Py joins on number ranges
# author: vladimir kulyukin
####################################

range1 = xrange(1, 6)

def join_number_range(separator, rng):
  return separator.join([str(x) for x in rng])

def range_tests(r):
  print join_number_range('*',     r)
  print join_number_range(' ** ',  r)
  print join_number_range('//',    r)

range_tests(range1)
